lib/cli/run.js

```diff
@@ -0,0 +1,403 @@
+'use strict';
+
+/**
+ * Command module for default ("run") command
+ */
+
+const path = require('path');
+const {loadConfig, findConfig} = require('./config');
+const debug = require('debug')('mocha:cli:run');
+const Mocha = require('../mocha');
+const {
+  list,
+  cwd,
+  runShowInterfaces,
+  runShowReporters,
+  handleExtensions,
+  handleFiles,
+  handleRequires,
+  validatePlugin,
+  runMocha
+} = require('./run-helpers');
+
+/**
+ * Logical option groups
+ */
+const GROUPS = {
+  FILES: 'File Handling',
+  FILTERS: 'Test Filters',
+  NODEJS: 'Node.js & V8',
+  OUTPUT: 'Reporting & Output',
+  RULES: 'Rules & Behavior',
+  CONFIG: 'Config Files'
+};
+
+/**
+ * Placeholder value to force config parsing function to run.
+ */
+const CONFIG_PATH_PLACEHOLDER = '__MOCHA_CONFIG_PATH__';
+
+exports.command = '$0 [spec..]';
```

Comment:
```js
const prog = path.basename(require('../package').bin.mocha);
exports.command = `Usage: ${prog} [options] [spec..]`;
```

If you use `$0`, then path to binary becomes part of usage (indeterminate).
Also, does it need to document `mocha debug`?

Reply:
`$0` is set to `mocha` via `yargs.scriptName()`

see above about `debug`.

Reply:
Guess that's a new feature...

Reply:
```js
exports.command = `Usage: $0 [options] [spec..]`;
```